### PR TITLE
Update naming-conventions.md

### PR DESCRIPTION
**Pull request by b1n0ys**
updated the naming convention for scala object

### CODE REVIEW

**Comment by SethTisue** [comment]
can you explain your thinking here? it seems me the original text is correct. perhaps we can make it clearer though.

**Comment by b1n0ys** [comment]
isn't it correct to have all objects in scala to be names with lower camel
case

unless they are companion Objects (in which case the name should be same as
the class and hence Upper camel case)
isn't that the exception case.


Thanks & Regards,
Binoy Sundaran

**Comment by SethTisue** [comment]
>isn't it correct to have all objects in scala to be names with lower camel case

no. one normally writes `object FooBar`, not `object fooBar`

**Comment by b1n0ys** [comment]
I guess then that's the confusion, you are explaining a case where we
specifically use 'object' keyword.

In the case we are defining a companion object or creating a singleton
object (which is a special case i would say)

but for normal instance Objects , people should follow lower camel casing

i am talking about the objects created like these.
val foo = new Foo


Thanks & Regards,
Binoy Sundaran

**Comment by SethTisue** [comment]
ah, I see. yes, there is sometimes confusion in conversations about Scala whether the word "object" refers to `object` specifically or just to any value (since although Scala has both value types and reference types, they're all considered "objects" in that, unlike primitive value types in Java, you can call methods on them)

**Comment by SethTisue** [comment]
later on the same page it says "Method, Value and variable names should be in lower camel case", which is intended to cover the `val foo = new Foo` case. but perhaps some wording and/or structural changes would help make this clearer

File: _style/naming-conventions.md [hunk 1]
```diff
@@ -47,6 +44,17 @@ This isn't common. Example:
     object inc {
       def apply(x: Int): Int = x + 1
     }
+ 
+Unless these are companion Objects, 
+in which case Object name should be same as class name(which then be upper camel case)
+ 
+    class Car (val year : Int, val miles : Int) {
```
Comment (jvican) [inline _style/naming-conventions.md]: If the template is empty, can we remove it?

File: _style/naming-conventions.md [hunk 2]
```diff
@@ -32,10 +32,7 @@ This mimics the Java naming convention for classes.
 
 ## Objects
 
-Object names are like class names (upper camel case).
-
-An exception is when mimicking a package or function.
-This isn't common. Example:
+Object should be named in lower camel case :
```
Comment (jvican) [inline _style/naming-conventions.md]: I think it's commonplace in the Scala community to always use upper camel case for `object`. I've rarely seen lower camel case for objects, I've only seen it for `package object`s.

**Comment by jvican** [comment]
Oh sorry, just read the last comments. I see the confusion. Probably the best thing is to rewrite the whole paragraph to avoid it.

**Comment by jvican** [comment]
@b1n0ys Do you have time to follow up on this? :smile:

**Comment by SethTisue** [comment]
closing for inactivity. 

see also https://contributors.scala-lang.org/t/what-happened-to-module/1790

**Pull request closed without merging**